server/routes/post-hire-status.ts

```diff
@@ -10,57 +11,109 @@ import keycloak from '../keycloak';
 import logger from '../logger';
 import { getParticipantByID } from '../services/participants';
 import { getAssignCohort } from '../services/cohorts';
+import { postHireStatuses } from '../constants';
 
 const router = express.Router();
 
+type postHireStatusBody = {
+  body: {
+    participantIds: number[]; // participant IDs getting their post hire status set/updated
+    status: postHireStatuses; //  String representing the status
+    data: {
+      graduationDate?: string; // Date set If status = successful
+      unsuccessfulCohortDate?: string; // Date set If status = unsuccessful
+    };
+  };
+  /** User setting the status */
+  user: {
+    user_id: string; // User id from keycloak
+    sub: string; // Additional user id from keycloak
+  };
+};
 // Apply setup user middleware
 router.use(applyMiddleware(keycloak.setupUserMiddleware()));
 
 router.post(
   '/',
   applyMiddleware(keycloak.allowRolesMiddleware('health_authority', 'employer')),
-  asyncMiddleware(async (req, res) => {
+  asyncMiddleware(async (req: postHireStatusBody, res) => {
     const { user_id: userId, sub: localUserId } = req.user;
     const { body } = req;
 
     const user = userId || localUserId;
     // Validate the request body
     await validate(ParticipantPostHireStatusSchema, body);
-    const { participantId } = body;
-    // Check participant exists
-    const [participant] = await getParticipantByID(participantId);
-    if (!participant) {
+    const { participantIds } = body;
+
+    const notParticipants: {
+      id: number;
+      valid: boolean;
+    }[] = (
+      await Promise.all(
+        participantIds.map(async (id) => ({ id, valid: (await getParticipantByID(id)).length > 0 }))
```

Comment:
Bit interesting here. I think it works fine, though would've been nice to have a service to just get multiple participants by ids.